From 8b5c87c2f0c85be5754b29904a4644c9c40a02bc Mon Sep 17 00:00:00 2001
From: caseif <mproncace@gmail.com>
Date: Sun, 22 Feb 2015 00:05:50 -0500
Subject: [PATCH] Remove final mod from EntityDamageEvent#getFinalDamage()


diff --git a/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java b/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java
index 9ef48c8..28f700e 100644
--- a/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java
+++ b/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java
@@ -141,7 +141,7 @@ public class EntityDamageEvent extends EntityEvent implements Cancellable {
      *
      * @return the amount of damage caused by the event
      */
-    public final double getFinalDamage() {
+    public double getFinalDamage() {
         double damage = 0;
         for (DamageModifier modifier : MODIFIERS) {
             damage += getDamage(modifier);
@@ -153,7 +153,7 @@ public class EntityDamageEvent extends EntityEvent implements Cancellable {
      * This method exists for legacy reasons to provide backwards
      * compatibility. It will not exist at runtime and should not be used
      * under any circumstances.
-     * 
+     *
      * @return the (rounded) damage
      */
     @Deprecated
@@ -202,7 +202,7 @@ public class EntityDamageEvent extends EntityEvent implements Cancellable {
      * This method exists for legacy reasons to provide backwards
      * compatibility. It will not exist at runtime and should not be used
      * under any circumstances.
-     * 
+     *
      * @param damage the new damage value
      */
     @Deprecated
-- 
1.9.4.msysgit.2

