From 9c709c47c320b88e9d20c4bd83da00f6945532ec Mon Sep 17 00:00:00 2001
From: caseif <caseif@caseif.net>
Date: Sun, 8 Mar 2015 14:18:53 -0400
Subject: [PATCH] Prevent InventoryClickEvent from throwing NPEs


diff --git a/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java b/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
index 28198b8..fb575e2 100644
--- a/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
+++ b/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
@@ -62,7 +62,7 @@ public class InventoryClickEvent extends InventoryInteractEvent {
         super(view);
         this.slot_type = type;
         this.rawSlot = slot;
-        this.whichSlot = view.convertSlot(slot);
+        this.whichSlot = -1;
         this.click = click;
         this.action = action;
     }
-- 
1.9.4.msysgit.2

