From 7764c7ed1550d954577c20f06f8bb3d5d2dca35a Mon Sep 17 00:00:00 2001
From: caseif <caseif@caseif.net>
Date: Sat, 27 Jun 2015 15:28:06 -0400
Subject: [PATCH] Add isValid()Z method to Event superclass


diff --git a/src/main/java/org/bukkit/event/Event.java b/src/main/java/org/bukkit/event/Event.java
index d8ce291..0071e8f 100644
--- a/src/main/java/org/bukkit/event/Event.java
+++ b/src/main/java/org/bukkit/event/Event.java
@@ -99,4 +99,15 @@ public abstract class Event {
     protected final boolean isPoreEvent() {
         return getClass().getName().startsWith("blue.lapis.pore.impl.event."); // TODO: Keep up to date
     }
+
+    /**
+     * Returns <code>true</code> if this {@link Event} should be called for
+     * its wrapped Sponge event, otherwise returns <code>false</code>.
+     *
+     * @return Whether this {@link Event} should be called for
+     *         its wrapped Sponge event
+     */
+    public boolean isValid() {
+        return true;
+    }
 }
-- 
1.9.4.msysgit.2

